File: Flutter - Non-Native/discover_destination/lib/service/users_service.dart
import 'package:discover_destination/model/user.dart';

class UsersService {
  List<User> users = List.of([User("ana", "pass"), User("ion", "pass")]);

  bool checkUser(User user) {
    return !users.where((u) => user.username == u.username && user.password == u.password).isEmpty;
  }
}

File: Android - Client-Server/client/discover_destination/app/src/main/java/com/example/discover_destination/service/UsersService.kt
package com.example.discover_destination.service

import com.example.discover_destination.domain.User
import javax.inject.Inject

class UsersService @Inject constructor() {
    private var users: MutableList<User> = mutableListOf(User("ana", "pass"), User("ion", "pass"))

    fun checkUser(user: User): Boolean {
        return !users.filter{u -> user.username == u.username && user.password == u.password}.isEmpty()
    }
}